dimod/binary_quadratic_model/model.py

```diff
@@ -0,0 +1,1099 @@
+            linear[v] *= -1
+
+            for u in adj[v]:
+                bias = adj[v][u]
+
+                adj[v][u] *= -1.
+                adj[u][v] *= -1.
+
+                linear[u] += bias
+
+                if (u, v) in quadratic:
+                    quadratic[(u, v)] *= -1.
+                elif (v, u) in quadratic:
+                    quadratic[(v, u)] *= -1.
+                else:
+                    raise RuntimeError("quadratic is missing an interaction")
+
+        else:
+            raise RuntimeError("Unexpected vartype")
+
+        try:
+            self._counterpart.flip_variable(v)
+        except AttributeError:
+            pass
+
+    def update(self, bqm):
+        """Update with the values from another binary quadratic model.
+
+        Args:
+            bqm (:class:`.BinaryQuadraticModel`):
+                A binary quadratic model. All of the biases are added to self.
+
+        """
+        self.add_variables_from(bqm.linear, vartype=bqm.vartype)
+        self.add_interactions_from(bqm.quadratic, vartype=bqm.vartype)
+        self.add_offset(bqm.offset)
+
+    def contract_variables(self, u, v):
+        """Asserts u, v are the same variable.
+
+        The resulting variable will be labeled as 'u'.
+
+        Args:
+            u (variable):
+                A variable in the binary quadratic model.
+
+            v (variable):
+                A variable in the binary quadratic model.
+
+        """
+        adj = self.adj
+
+        if u not in adj:
+            raise ValueError("{} is not a variable in the binary quadratic model".format(u))
+        if v not in adj:
+            raise ValueError("{} is not a variable in the binary quadratic model".format(v))
+
+        # if there is an interaction between u, v it becomes linear for u
+        if v in adj[u]:
+            if self.vartype is Vartype.BINARY:
+                self.add_variable(u, adj[u][v])
+            elif self.vartype is Vartype.SPIN:
+                self.add_offset(adj[u][v])
+            else:
+                raise RuntimeError("unexpected vartype")
+            self.remove_interaction(u, v)
+
+        # all of the interactions that v has become interactions for u
+        neighbors = list(adj[v])
+        for w in neighbors:
+            self.add_interaction(u, w, adj[v][w])
+            self.remove_interaction(v, w)
+
+        # finally remove v
+        self.remove_variable(v)
+
+
+###################################################################################################
+# transformations
+###################################################################################################
+
+    def relabel_variables(self, mapping, copy=True):
+        """Relabel the variables according to the given mapping.
+
+        Args:
+            mapping (dict): a dict mapping the current variable labels
+                to new ones. If an incomplete mapping is provided,
+                variables will keep their labels
```

Comment:
"_unmapped variables_ will keep their labels" to clarify that the mapping will still be executed.

Reply:
Fixed